content/blog/2022-07-26-schema-evolution-glue-msk/index.md

```diff
@@ -0,0 +1,861 @@
+...
+[WARNING] 
+org.apache.kafka.common.errors.SerializationException: Error deserializing key/value for partition unicorn-ride-request-topic-0 at offset 100. If needed, please seek past the record to continue consumption.
+Caused by: com.amazonaws.services.schemaregistry.exception.AWSSchemaRegistryException: Exception occurred while de-serializing Avro message
+    at com.amazonaws.services.schemaregistry.deserializers.avro.AvroDeserializer.deserialize (AvroDeserializer.java:103)
+...
+Caused by: org.apache.avro.AvroTypeException: Found cloud.localstack.demos.gluemsk.schema.UnicornRideRequest, expecting cloud.localstack.demos.gluemsk.schema.UnicornRideRequest, missing required field customer
+    at org.apache.avro.io.ResolvingDecoder.doAction (ResolvingDecoder.java:308)
+...
+```
+
+### 4. Updated consumers
+Finally, we will update our consumer such that it is compatible to the new version of the schema.
+
+Similar to the previous producer scenarios, we create a new consumer by executing the following steps:
+- Copy the `consumer` directory and rename it to `consumer-2`.
+- Set a new artifact ID in the `pom.xml` of the module:
+  ```xml
+    ...
+    <artifactId>consumer-2</artifactId>
+    ...
+  ```
+- Add the new module to the root `pom.xml`:
+  ```xml
+  <modules>
+    <module>producer</module>
+    <module>producer-2</module>
+    <module>producer-3</module>
+    <module>consumer</module>
+    <module>consumer-2</module>
+  </modules>
+  ```
+- Replace the `unicorn_ride_request_v1.avsc` with the new version used by `producer-2` (`unicorn_ride_request_v2.avsc`).
+- Re-generate the Java classes for the schema:
+  ```bash
+  mvn clean generate-sources
+  ```
+
+[The complete module can be found in our samples repository (along with the rest of the code of this tutorial).](https://github.com/localstack/localstack-pro-samples/blob/cd023a84a3b473984e9c34053d4feb7de8e038c1/glue-msk-schema-registry/consumer-2/)
+
+Our new consumer, based on the latest version of the schema, will be able to successfully consume the records published by the new producer:
+```bash
+mvn -pl consumer-2 exec:java -Dexec.args="--bootstrap-servers localhost:4511"
+```
+
+# Conclusion
+Apache Kafka is used as the core messaging system in complex environments, with independent producers and consumers.
+The individual development of these components make it hard to coordinate and evolve data schemas over time.
+Using the AWS Glue Schema Registry can help you to prevent the usage of incompatible schemas.
+
+With LocalStack, emulating Amazon Managed Streaming for Kafka and AWS Glue Schema Registry, you can develop and test the next evolution of your data schema locally on your own machine.
```

Comment:
```suggestion
Apache Kafka is the core messaging system in complex environments with independent producers and consumers. The individual development of these components makes it hard to coordinate and evolve data schemas over time. Using the AWS Glue Schema Registry can help you to prevent the usage of incompatible schemas.
		
With LocalStack, emulating Amazon Managed Streaming for Kafka and AWS Glue Schema Registry, you can develop and test the next evolution of your data schema locally on your machine.
```

Reply:
As above, I think this change only removes newlines.